compiler/rustc_parse/src/parser/item.rs

```diff
@@ -700,6 +705,50 @@ impl<'a> Parser<'a> {
         Ok(ItemKind::Impl(Impl { generics, of_trait, self_ty, items: impl_items, constness }))
     }
 
+    fn parse_item_auto_impl(&mut self, attrs: &mut AttrVec) -> PResult<'a, ItemKind> {
+        let safety = self.parse_safety(Case::Sensitive);
+        self.psess.gated_spans.gate(sym::supertrait_auto_impl, self.token.span);
+        debug_assert!(self.eat_keyword(exp!(Auto)));
+        debug_assert!(self.eat_keyword(exp!(Impl)));
```

Comment:
This won't eat these keywords / advance the parser state in release builds of the compiler.